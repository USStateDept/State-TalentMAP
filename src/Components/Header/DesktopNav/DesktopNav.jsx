import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import { USER_PROFILE } from '../../../Constants/PropTypes';
import GlossaryIcon from '../GlossaryIcon';
import NavLink from '../NavLink';
import AccountDropdown from '../../AccountDropdown/AccountDropdown';
import InteractiveElement from '../../InteractiveElement';

const DesktopNav = ({ isLoggedIn, shouldShowSearchBar,
userProfile, logout, toggleSearchVisibility }) => (
  <div className="navigation-container">
    <div className="nav-link-container header-nav-desktop desktop-nav-only">
      {
        isLoggedIn &&
        <div className={`header-nav-link-container ${shouldShowSearchBar ? 'is-highlighted' : 'is-not-highlighted'}`}>
          <div className="header-nav-link">
            <div className="header-nav-link-text search-text">
              <InteractiveElement
                type="span"
                onClick={toggleSearchVisibility}
              >
                <FontAwesome name="search" /> Search
              </InteractiveElement>
            </div>
          </div>
        </div>
      }
      <NavLink link="/" title="Home" navLinkClass="home-text" />
      <div className="header-nav-link-container">
        <div className="header-nav-link">
          <div className="header-nav-link-text about-text">
            <a href="https://github.com/18F/State-TalentMAP">About</a>
          </div>
        </div>
      </div>
    </div>
    <div className="header-nav-desktop desktop-nav-only account-notification-container">
      <div className="header-nav-link-container account-container">
        <div className="header-nav-link">
          {
            isLoggedIn &&
            <AccountDropdown
              userProfile={userProfile}
              logoutRequest={logout}
            />
          }
        </div>
      </div>
    </div>
    <div className="header-nav-desktop desktop-nav-only account-notification-container">
      <div className="header-nav-link-container notifications-container">
        <div className="header-nav-link">
          {
            isLoggedIn &&
              <span>
                <GlossaryIcon />
              </span>
          }
        </div>
      </div>
    </div>
  </div>
);

DesktopNav.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  shouldShowSearchBar: PropTypes.bool.isRequired,
  userProfile: USER_PROFILE,
  logout: PropTypes.func.isRequired,
  toggleSearchVisibility: PropTypes.func.isRequired,
};

DesktopNav.defaultProps = {
  userProfile: {},
};

export default DesktopNav;
