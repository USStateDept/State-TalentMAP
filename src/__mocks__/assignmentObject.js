const assignmentObject = {
  start_date: '10-10-2016',
  end_date: '10-10-2018',
  estimated_end_date: '10-10-2018',
  position: {
    skill: 'CONSTRUCTION ENGINEERING (6218)',
    bureau: '(NEA) BUREAU NEAR EASTERN AFFAIRS',
    position_number: '58348065',
    title: 'PROJECT DIRECTOR',
    post: {
      location: {
        id: 4,
        country: 'United States',
        code: '500',
        city: 'Washington',
        state: 'DC',
      },
    },
  },
};

export default assignmentObject;
