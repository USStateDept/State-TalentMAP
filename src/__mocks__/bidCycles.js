export default [
  {
    id: 1,
    name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
    cycle_start_date: '2017-12-13T15:51:10.980554Z',
    cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
    cycle_end_date: '2018-06-13T15:51:10.980554Z',
    active: true,
  },
];
