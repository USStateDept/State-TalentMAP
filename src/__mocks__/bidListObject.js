const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: 'Curacao',
        },
      },
      status: 'approved',
      submission_date: '2017-10-31',
      update_date: '2017-10-31',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: 'Freetown, Sierra Leone',
        },
      },
      status: 'closed',
      submission_date: '2017-10-31',
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: 'The Hague, Netherlands',
        },
      },
      status: 'closed',
      submission_date: null,
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
