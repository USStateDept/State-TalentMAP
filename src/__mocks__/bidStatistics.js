const bidStatistics = {
  id: 1,
  name: 'Demo BidCycle 2018-01-10 15:52:20.583434',
  cycle_start_date: '2017-10-10T00:00:00Z',
  cycle_deadline_date: '2018-02-10T00:00:00Z',
  cycle_end_date: '2018-04-10T00:00:00Z',
  total_positions: 251,
  available_positions: 240,
  available_domestic_positions: 81,
  available_international_positions: 159,
  total_bids: 40,
  total_bidders: 4,
  in_panel_bidders: 0,
  approved_bidders: 1,
  bidding_days_remaining: 30,
};

export const bidStatisticsList = {
  results: [
    bidStatistics,
  ],
};

export default bidStatistics;
