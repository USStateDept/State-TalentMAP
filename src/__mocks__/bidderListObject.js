const bidderListObject = {
  count: 6,
  next: null,
  previous: null,
  results: [
    {
      id: 5,
      current_assignment: '(active) Kara Batisak at [58312265] PROJECT DIRECTOR (Guatemala, Guatemala)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 1,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Kara Batisak',
          draft: 6,
          submitted: 3,
          handshake_offered: 1,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'batisak',
        email: 'batisak@state.gov',
        first_name: 'Kara',
        last_name: 'Batisak',
      },
      language_qualifications: [],
    },
    {
      id: 3,
      current_assignment: '(active) John Doe at [10181012] INTERNATIONAL RELATIONS OFF (Singapore, Singapore)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'doej',
        email: 'doej@talentmap.us',
        first_name: 'John',
        last_name: 'Doe',
      },
      language_qualifications: [],
    },
    {
      id: 1,
      current_assignment: '(active) Guest McGuestson at [58312870] CONSTRUCTION MANAGER (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '03',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'guest',
        email: 'guest@state.gov',
        first_name: 'Guest',
        last_name: 'McGuestson',
      },
      language_qualifications: [],
    },
    {
      id: 6,
      current_assignment: '(active) Tarek Rehman at [56457009] ASST REGIONAL SECURITY OFCR (Adana, Turkey)',
      skills: [
        {
          id: 20,
          cone: null,
          code: '2501',
          description: 'SECURITY',
        },
      ],
      grade: '04',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 2,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Tarek Rehman',
          draft: 5,
          submitted: 2,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'rehmant',
        email: 'rehmant@state.gov',
        first_name: 'Tarek',
        last_name: 'Rehman',
      },
      language_qualifications: [],
    },
    {
      id: 2,
      current_assignment: '(active) Administrator TalentMAP at [58312869] PROJECT DIRECTOR (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'admin',
        email: 'admin@talentmap.us',
        first_name: 'Administrator',
        last_name: 'TalentMAP',
      },
      language_qualifications: [],
    },
    {
      id: 4,
      current_assignment: '(active) Jenny Townpost at [10026129] INTERNATIONAL RELATIONS OFF (Port Moresby, Papua New Guinea)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 4,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Jenny Townpost',
          draft: 3,
          submitted: 4,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'townpostj',
        email: 'townpostj@state.gov',
        first_name: 'Jenny',
        last_name: 'Townpost',
      },
      language_qualifications: [],
    },
  ],
};

export default bidderListObject;
