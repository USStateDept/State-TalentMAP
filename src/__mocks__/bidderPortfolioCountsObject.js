const bidderPortfolioCountsObject = {
  all_clients: 100,
  bidding_clients: 50,
  in_panel_clients: 20,
  on_post_clients: 30,
  bidding_no_handshake: 10,
};

export default bidderPortfolioCountsObject;
