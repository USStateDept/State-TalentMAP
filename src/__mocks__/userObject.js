export const bidderUserObject = {
  id: 1,
  grade: '03',
  skills: [
    {
      id: 63,
      cone: null,
      code: '6218',
      description: 'CONSTRUCTION ENGINEERING',
    },
    {
      id: 40,
      cone: null,
      code: '6211',
      description: 'COMPUTER SCIENCE',
    },
  ],
  user: {
    initials: 'JD',
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
    username: 'jdoe',
  },
  display_name: 'John',
  initials: 'JD',
  is_cdo: false,
  cdo: {
    username: 'shadtrachl',
    initials: 'LS',
    first_name: 'Leah',
    last_name: 'Shadtrach',
    email: 'shadtrachl@state.gov',
  },
  bid_statistics: [
    {
      draft: 1,
      submitted: 1,
    },
  ],
  favorite_positions: [
    {
      id: 1,
      representation: '[00003026] OMS (COM) (Freetown, Sierra Leone)',
    },
    {
      id: 4,
      representation: '[00180000] OMS (DCM) (Addis Ababa, Ethiopia)',
    },
  ],
};

export const cdoUserObject = Object.assign({}, bidderUserObject, { cdo: null, is_cdo: true });
