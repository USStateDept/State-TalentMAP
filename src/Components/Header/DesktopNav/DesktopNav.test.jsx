import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DesktopNav from './DesktopNav';
import { bidderUserObject } from '../../../__mocks__/userObject';

describe('DesktopNav', () => {
  beforeEach(() => {
    process.env.PUBLIC_URL = '/talentmap/';
  });

  it('is defined', () => {
    const wrapper = shallow(
      <DesktopNav
        isLoggedIn
        shouldShowSearchBar
        userProfile={bidderUserObject}
        logout={() => {}}
        toggleSearchVisibility={() => {}}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot when logged in', () => {
    const wrapper = shallow(
      <DesktopNav
        isLoggedIn
        shouldShowSearchBar
        userProfile={bidderUserObject}
        logout={() => {}}
        toggleSearchVisibility={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when logged out', () => {
    const wrapper = shallow(
      <DesktopNav
        isLoggedIn={false}
        shouldShowSearchBar
        userProfile={bidderUserObject}
        logout={() => {}}
        toggleSearchVisibility={() => {}}
      />,
      );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
