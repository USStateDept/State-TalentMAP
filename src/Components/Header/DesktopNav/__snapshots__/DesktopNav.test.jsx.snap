// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNav matches snapshot when logged in 1`] = `
<div
  className="navigation-container"
>
  <div
    className="nav-link-container header-nav-desktop desktop-nav-only"
  >
    <div
      className="header-nav-link-container is-highlighted"
    >
      <div
        className="header-nav-link"
      >
        <div
          className="header-nav-link-text search-text"
        >
          <InteractiveElement
            className=""
            onClick={[Function]}
            type="span"
          >
            <FontAwesome
              name="search"
            />
             Search
          </InteractiveElement>
        </div>
      </div>
    </div>
    <withRouter()
      link="/"
      navLinkClass="home-text"
      title="Home"
    />
    <div
      className="header-nav-link-container"
    >
      <div
        className="header-nav-link"
      >
        <div
          className="header-nav-link-text about-text"
        >
          <a
            href="https://github.com/18F/State-TalentMAP"
          >
            About
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="header-nav-desktop desktop-nav-only account-notification-container"
  >
    <div
      className="header-nav-link-container account-container"
    >
      <div
        className="header-nav-link"
      >
        <AccountDropdown
          logoutRequest={[Function]}
          shouldDisplayName={true}
          userProfile={
            Object {
              "bid_statistics": Array [
                Object {
                  "draft": 1,
                  "submitted": 1,
                },
              ],
              "cdo": Object {
                "email": "shadtrachl@state.gov",
                "first_name": "Leah",
                "initials": "LS",
                "last_name": "Shadtrach",
                "username": "shadtrachl",
              },
              "display_name": "John",
              "favorite_positions": Array [
                Object {
                  "id": 1,
                  "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
                },
                Object {
                  "id": 4,
                  "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
                },
              ],
              "grade": "03",
              "id": 1,
              "initials": "JD",
              "is_cdo": false,
              "skills": Array [
                Object {
                  "code": "6218",
                  "cone": null,
                  "description": "CONSTRUCTION ENGINEERING",
                  "id": 63,
                },
                Object {
                  "code": "6211",
                  "cone": null,
                  "description": "COMPUTER SCIENCE",
                  "id": 40,
                },
              ],
              "user": Object {
                "email": "doej@state.gov",
                "first_name": "John",
                "initials": "JD",
                "last_name": "Doe",
                "username": "jdoe",
              },
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="header-nav-desktop desktop-nav-only account-notification-container"
  >
    <div
      className="header-nav-link-container notifications-container"
    >
      <div
        className="header-nav-link"
      >
        <span>
          <Connect(GlossaryIcon) />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`DesktopNav matches snapshot when logged out 1`] = `
<div
  className="navigation-container"
>
  <div
    className="nav-link-container header-nav-desktop desktop-nav-only"
  >
    <withRouter()
      link="/"
      navLinkClass="home-text"
      title="Home"
    />
    <div
      className="header-nav-link-container"
    >
      <div
        className="header-nav-link"
      >
        <div
          className="header-nav-link-text about-text"
        >
          <a
            href="https://github.com/18F/State-TalentMAP"
          >
            About
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="header-nav-desktop desktop-nav-only account-notification-container"
  >
    <div
      className="header-nav-link-container account-container"
    >
      <div
        className="header-nav-link"
      />
    </div>
  </div>
  <div
    className="header-nav-desktop desktop-nav-only account-notification-container"
  >
    <div
      className="header-nav-link-container notifications-container"
    >
      <div
        className="header-nav-link"
      />
    </div>
  </div>
</div>
`;
